Java/syntax_test_java.java

```diff
@@ -1,25 +1,96 @@
 // SYNTAX TEST "Packages/Java/Java.sublime-syntax"
 
 package apple;
-// <- source.java meta.package.java keyword.other.package.java
-//      ^ meta.package.java support.other.package.java
-//           ^ meta.package.java punctuation.terminator.java
+// <- source.java meta.package-declaration.java keyword.other.package.java
+//      ^^^^^ entity.name.package.java
+//           ^ punctuation.terminator.java
+
+package com.example.apple;
+//^^^^^^^^^^^^^^^^^^^^^^^ meta.package-declaration.java
+//      ^^^^^^^^^^^^^^^^^ entity.name.package.java
+//         ^ punctuation.accessor.dot.java
+//                 ^ punctuation.accessor.dot.java
+//                       ^ punctuation.terminator.java
 
 import a.b.Class;
 // <- meta.import.java keyword.other.import.java
-//     ^ meta.import.java support.class.import.java
+//     ^^^^^^^^^ meta.path.java
+//     ^ support.type.package.java
+//      ^ punctuation.accessor.dot.java
+//       ^ support.type.package.java
+//        ^ punctuation.accessor.dot.java
+//         ^^^^^ support.class.import.java
+//              ^ punctuation.terminator.java
+
+import com.google
+//     ^^^^^^^^^^ meta.import.java meta.path.java
 //        ^ punctuation.accessor.dot.java
-//              ^ meta.import.java punctuation.terminator.java
+//         ^^^^^^ support.type.package.java
+  .common.collect
+//^ punctuation.accessor.dot.java
+//       ^ punctuation.accessor.dot.java
+//        ^^^^^^^ support.type.package.java
+//^^^^^^^^^^^^^^^ meta.import.java meta.path.java
+  .ListMultimap;
+//^ punctuation.accessor.dot.java
+// ^^^^^^^^^^^^ support.class.import.java
+//^^^^^^^^^^^^^ meta.import.java meta.path.java
+//             ^ punctuation.terminator.java
+
+import no.terminator
+// <- meta.import.java keyword.other.import.java
+
+import static no.terminator
+// <- meta.import.java keyword.other.import.java
+
+import
+// <- meta.import.java keyword.other.import.java
+
+import static
+// <- meta.import.java keyword.other.import.java
+
+import java.net.URL;
+// <- meta.import.java keyword.other.import.java
+//^^^^^^^^^^^^^^^^^ meta.import.java
+//     ^^^^^^^^^^^^ meta.path.java
+//              ^^^ support.class.import.java
+//     ^^^^ support.type.package.java
+//         ^ punctuation.accessor.dot.java
+//          ^^^ support.type.package.java
+//             ^ punctuation.accessor.dot.java
+//                 ^ punctuation.terminator.java
+
+import java.util.*;
+//^^^^^^^^^^^^^^^^ meta.import.java
+//     ^^^^^^^^^^^ meta.path.java
+//     ^^^^ support.type.package.java
+//         ^ punctuation.accessor.dot.java
+//          ^^^^ support.type.package.java
+//              ^ punctuation.accessor.dot.java
+//               ^ keyword.operator.wildcard.asterisk.java
 
 import static a.b.Class.fooMethod;
+//^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ meta.import.java
```

Comment:
not sure if it's possible to structure the meta classes to make it easy for a script to pull out all the static and non static imports:

    view.find_by_selector("meta.import meta.path") # doesn't get static methods.

Reply:
I definitely won't get methods (and constants) as `meta.path` can only be part of the class id. We may add something like `meta.imported` for everything that comes after the keywords.